CONTRIBUTING.md

```diff
@@ -1,61 +1,47 @@
-[prometheus-k8s operator][Prometheus operator], although that is not
-necessary, as [Alertmanager's HTTP API][Alertmanager API browser] could be
-[used](https://github.com/prometheus/alertmanager/issues/437#issuecomment-263413632)
-instead.
-
-## Known issues
-1. Adding multiple receivers of the same type (e.g. PagerDuty) is not very scalable, due to the
-   nature of the `juju config` command. This is likely to improve in the future by using integrator
-   charms.
-
-## Bugs and pull requests
-- Generally, before developing enhancements to this charm, you should consider
-  explaining your use case.
-- If you would like to chat with us about your use-cases or proposed
-  implementation, you can reach us at
-  [Canonical Mattermost public channel](https://chat.charmhub.io/charmhub/channels/charm-dev)
-  or [Discourse](https://discourse.charmhub.io/).
-- All enhancements require review before being merged. Apart from
-  code quality and test coverage, the review will also take into
-  account the resulting user experience for Juju administrators using
-  this charm.
+
+## Overview
+
+This documents explains the processes and practices recommended for
+contributing enhancements or bug fixing to the Alertmanager Charmed Operator.
+
+The intended use case of this operator is to be deployed as part of the
+[COS Lite] bundle, although that is not necessary.
+
 
 ## Setup
 
 A typical setup using [snaps](https://snapcraft.io/) can be found in the
 [Juju docs](https://juju.is/docs/sdk/dev-setup).
 
+
 ## Developing
 
-Use your existing Python 3 development environment or create and
-activate a Python 3 virtualenv
+- Prior to getting started on a pull request, we first encourage you to open an
+  issue explaining the use case or bug.
+  This gives other contributors a chance to weigh in early in the process.
+- To author PRs you should know [what is jujuj](https://juju.is/#what-is-juju)
+  and [how operators are written](https://juju.is/docs/sdk).
+- The best way to get a head start is to join the conversation on our
+  [Mattermost channel] or [Discourse].
+- All enhancements require review before being merged. Besides the
+  code quality and test coverage, the review will also take into
+  account the resulting user experience for Juju administrators using
+  this charm. To be able to merge you would have to rebase
+  onto the `main` branch. We do this to avoid merge commits and to have a
+  linear Git history.
+- We use [`tox`](https://tox.wiki/en/latest/#) to manage all virtualenvs for
+  the development lifecycle.
 
-```shell
-virtualenv -p python3 venv
-source venv/bin/activate
-```
 
-Install the development requirements
+### Testing
+Unit tests are written with the Operator Framework [test harness] and
+integration tests are written using [pytest-operator] and [python-libjuju].
 
-```shell
-pip install -r requirements.txt
-```
+All default tests can be executed by running `tox` without arguments.
```

Comment:
This sentence doesn't really convey any information. If you run `tox` without arguments, you will run the default tests, because that is what the default tests are :-)

I would recommend detailing what will run when executing `tox`. Are there any external dependencies that the tox file will not setup? Will it make any unsafe changes to the system running the tests?

Reply:
Fixed in next commit.